aicsimageio/aics_image.py

```diff
@@ -0,0 +1,418 @@
+              range function. D should be present in the out_orientation.
+            * D=slice(...): D is Dimension letter and slice is the standard Python
+              slice function. D should be present in the out_orientation.
+
+        Returns
+        -------
+        data: numpy array
+            The image data with the specified dimension ordering.
+
+        Examples
+        --------
+        Specific index selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... c1 = img.get_image_data("ZYX", C=1)
+
+        List of index selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... first_and_second = img.get_image_data("CZYX", C=[0, 1])
+
+        Tuple of index selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... first_and_last = img.get_image_data("CZYX", C=(0, -1))
+
+        Range of index selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... first_three = img.get_image_dask_data("CZYX", C=range(3))
+
+        Slice selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... every_other = img.get_image_data("CZYX", C=slice(0, -1, 2))
+
+        Notes
+        -----
+        * If a requested dimension is not present in the data the dimension is
+          added with a depth of 1.
+
+        See `aicsimageio.transforms.reshape_data` for more details.
+        """
+        pass
+
+    @property
+    def metadata(self) -> Any:
+        """
+        Passthrough to the base image reader metadata property.
+        For more information, see the specific image format reader you are using
+        for details on it's metadata property.
+        """
+        pass
+
+    def get_channel_names(self, scene: int = 0) -> List[str]:
+        """
+        Attempts to use the available metadata for the image to return the channel
+        names for the image.
+
+        If no channel metadata is available, returns a list of string indicies for each
+        channel in the image.
+
+        Parameters
+        ----------
+        scene: int
+            The scene index to return channel names for.
+
+        Returns
+        -------
+        channel_names: List[str]
+            List of strings representing channel names.
+        """
+        pass
+
+    def get_physical_pixel_size(self, scene: int = 0) -> Tuple[float]:
+        """
+        Attempts to use the available metadata for the image to return the physical
+        pixel sizes for the image.
+
+        If no pixel metadata is available, returns `1.0` for each spatial dimension.
+
+        Paramaters
+        ----------
+        scene: int
+            The scene index to return physical pixel sizes for.
+
+        Returns
+        -------
+        sizes: Tuple[float]
+            Tuple of floats representing physical pixel sizes for dimensions X, Y, Z
+            (in that order).
+        """
+        pass
+
+    def save(
```

Comment:
just wondering, is anyone using this?

Reply:
Will worry about including this later after writers have been finished.